pyproject.toml

```diff
@@ -3,64 +3,74 @@ requires = ["pdm-backend"]
 build-backend = "pdm.backend"
 
 [tool.pdm.build]
-includes = [
-    "stac_model",
-]
+includes = ["stac_model"]
 
 [project]
 authors = [
-    {name = "Ryan Avery", email = "ryan@wherobots.com"},
-    {name = "Francis Charette-Migneault", email = "francis.charette-migneault@crim.ca"},
+  { name = "Ryan Avery", email = "ryan@wherobots.com" },
+  { name = "Francis Charette-Migneault", email = "francis.charette-migneault@crim.ca" },
 ]
-license = {text = "Apache Software License 2.0"}
+license = { text = "Apache Software License 2.0" }
 requires-python = "<4.0,>=3.10"
 dependencies = [
-    "typer<1.0.0,>=0.9.0",
-    "rich>=13.7.0,<15.0.0",
-    "pydantic<3.0.0,>=2.6.3",
-    "pydantic-core<3,>=2",
-    "pystac<2.0.0,>=1.9.0",
-    "shapely<3,>=2",
-    "jsonschema<5.0.0,>=4.21.1",
-    "pip>=25.0.0",
+  "typer<1.0.0,>=0.9.0",
+  "rich>=13.7.0,<15.0.0",
+  "pydantic<3.0.0,>=2.6.3",
+  "pydantic-core<3,>=2",
+  "pystac<2.0.0,>=1.9.0",
+  "shapely<3,>=2",
+  "jsonschema<5.0.0,>=4.21.1",
+  "pip>=25.0.0",
+]
+optional-dependencies.torch = [
+  "torch>=2.8.0.dev,<2.9.0; python_version>='3.11'",
```

Comment:
this should be torch<=2.8. versions beyond 2.8 could change the export API so we should be more conservative here.

Reply:
These new functions only work in 2.8 though so should we then pin to torch==2.8.0?